### ISSUE TITLE
RSpec spies not working with AnyArgMatcher

### ISSUE THREAD

**Issue by wvanbergen** [technicalpickles/rspec-spies#1]
RSpec spies does not work with an AnyArgMatcher, i.e.: the following will fail when it shouldn't:

```
s = "hello world"
s.stub(:slice)
s.slice(1,2)
s.should have_received(:slice) # fails
s.should have_received(:slice).with(any_args) # fails
```

Unfortunately, I quickly lost my way in the RSpec source code, so I was not able to fix this issue myself. Hopefully, you have more luck doing this. I will also open a ticket in the RSpec lighthouse, because rspec-spies is included in RSpec already, which I think is great.

**Comment by wvanbergen** [comment]
The RSpec ticket can be found here: https://rspec.lighthouseapp.com/projects/5645/tickets/938

**Comment by jamesarosen** [comment]
I believe this should be fixed in a7fc78797dcd6d10578e58fb1b243e75b2b52e36. Please reopen if it isn't.
